src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md

```diff
@@ -0,0 +1,110 @@
+date: 2024-04-08
+authors: ["sumit-shinde"]
+image: 
+tags:
+   - node-red
+   - post
+   - telegram
+---
+
+Integrating messaging applications or services in home automation is becoming common. Telegram is a popular choice among those messaging platforms. Today, in this guide, we will demonstrate how to integrate Telegram with Node-RED applications for sending and receiving messages
+
+<!--more-->
+
+## Creating a bot in Telegram
+
+- Navigate to your Telegram application and click on the top right search icon, make sure you have a Telegram account and are logged in with it.
+- Then type “botFather” and you'll see many accounts. Make sure to select the one with the blue verified tick.
+
+!["screenshot displaying searching for botFather bot for creating custom bot"](./images/sending-telegram-with-node-red-botfather.png "screenshot displaying searching for botFather bot for creating custom bot"){data-zoomable}
+
+- After that, you'll see the chat interface. Type `/newbot` and press enter to send the command.
+- The bot will ask for a name, which will be the display name of your bot. You can choose any name you like.
+- Next, it will ask for a unique username for your bot. The username should not include spaces and must end with 'bot' for example telegram_bot or telegrambot
+- Once you have entered a correct and unique username, you’ll receive a message confirming that you have successfully created the bot. It will contain the secret access token and a link to initiate your bot.
+- Click on the provided link. You'll be directed to the chat interface between you and the created bot, with a start button at the bottom. Click on that button to activate your bot.
+
+!["screenshot displaying chat interface with start buttom for activating your bot"](./images/sending-telegram-with-node-red-activating-bot.png "screenshot displaying chat interface with start buttom for activating your bot"){data-zoomable}
+
+## Obtaining your Telegram Chat ID
+
+Telegram chat ID is a unique identifier for a chat or a group in Telegram, which is required for sending and receiving messages. In this section, you'll learn how to obtain your own and your group's Telegram chat IDs.
+
+### Obtaining your Telegram Chat ID
+
+- Open your Telegram app Click on the top right search icon and type “Get My ID”
+
+!["screenshot displaying searching "Get My ID" bot for obtaining chat id"](./images/sending-telegram-with-node-red-getmyid.png "screenshot displaying searching "Get My ID" bot for obtaining chat id"){data-zoomable}
+
+- Click on the first account you'll see, the chat interface will open, and type `/start` and press enter.
+- After that, you’ll receive a message containing your Chat ID and User ID.
+
+### Obtaining your Telegram group Chat ID
+
+- Add `@getmyid_bot` to the group on which you want to send or receive messages using Node-RED.
+- Once the bot joins the group, the bot will send the chat ID.
+
+## Installing custom node
+
+- Switch to the "Install" tab
```

Comment:
Missing a step or two.  Tell the reader to switch over to FlowFuse and log into your node-red instance or something like that.